players/g1_player.py

```diff
@@ -363,19 +355,32 @@ def next_target(self, curr_loc: Tuple[float, float], goal: Point2D, conf: float)
                 return next_sp.point
 
             # Add adjacent points to heap
-            next_p_after_rolling = next_p if next_sp.previous is None else roll(next_sp.previous.point, next_p, constants.extra_roll)
+            if next_sp.previous is None:
+                next_p_after_rolling = next_p
+            elif is_in_sand_trap(next_p, self.sand_trap_matlab_polys, cache=self.map_points_in_sand_trap):
+                next_p_after_rolling = next_p
+            elif np.linalg.norm(np.array(self.goal) - np.array(next_sp.previous.point)) < 20:
+                if not is_in_sand_trap(next_sp.previous.point, self.sand_trap_matlab_polys, cache=self.map_points_in_sand_trap):
+                    next_p_after_rolling = next_p
```

Comment:
What scenario does this branch handle? When we are shooting within 20 meters of the goal and the shooting point is not in a sand trap, we don't apply rolling.

What if these conditions are all satisfied, and in addition we are actually aiming for a point more than 20 meters away? Consider the case when there's water between where we are and the goal, so we'll not make a putter shot.